coala_quickstart/generation/Project.py

```diff
@@ -25,3 +26,66 @@ def get_project_dir(printer):
             printer.print("Please enter a valid directory", color="blue")
 
     return project_dir
+
+
+def get_popular_languages(file_paths):
+    """
+    Identifies the most popular languages in the user's project directory
+    from the files matched from the given glob expression.
+
+    :param file_paths:
+        A list of absolute file paths in the user's project directory.
+    :return:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    results = {}
+    for ext in exts:
+        results[exts[ext]] = 0
+    results["Unknown"] = 0
+
+    for file_path in file_paths:
+        name, ext = os.path.splitext(file_path)
+        if ext in exts:
+            results[exts[ext]] += 1
+        else:
+            results["Unknown"] += 1
+    if len(file_paths):
+        for ext in results:
+            results[ext] = (100.0 * results[ext]) / len(file_paths)
+
+    languages = sorted(
+        results.items(),
+        key=operator.itemgetter(1),
+        reverse=True)
+    used_languages = []
+    append_list = []
+    for lang, percent in languages:
+        if lang == "Unknown":
+            append_list += [(lang, int(percent))]
+            continue
+        if percent == 0:
+            break
+        else:
+            used_languages += [(lang, int(percent))]
```

Comment:
I'm not simply sorting - the first `n - 1` elements are sorted, but I'm moving the Unknown to the bottom irrespective of its percentage (from what I see Unknown takes about 25% consistently from random files).

Also I'm not sure about the threshold idea - in a large project, there may be just one `README.rst`, but we should still return bears related to RST. So I'm still for the absolute zero percentage thing.

Agreed on merging the Unknown if with the else statement (but it still needs to be in its own if statement)

Also I'm converting to int again cause `39.89321471201%` doesn't look good :) Or did you mean differently in your question?